model-optimizer/extensions/middle/ONNXResize10ToInterpolateV4.py

```diff
@@ -0,0 +1,138 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging as log
+import numpy as np
+
+from extensions.ops.Cast import Cast
+from extensions.ops.activation_ops import Floor
+from extensions.ops.elementwise import Add, Mul
+from extensions.ops.interpolate import Interpolate
+from mo.front.common.layout import get_depth_dim, get_height_dim, get_width_dim
+from mo.front.common.partial_infer.utils import int64_array, float_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.passes.convert_data_type import data_type_str_to_np
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.const import Const
+from mo.ops.shape import Shape
+from mo.ops.strided_slice import StridedSlice
+
+
+def replace_resize(graph: Graph, resize: Node):
+    log.debug("Converting of ONNX Resize-10 to Interpolate-4 "
+              "is triggered for node {}.".format(resize.soft_get('name', resize.id)))
```

Comment:
Why can't we replace ONNX Resize with Interpolate on the front stage?

Reply:
No, because to replace ONNX Resize-10 we need to know input shapes.